aws/logs_monitoring/parsing.py

```diff
@@ -425,11 +425,14 @@ def parse_service_arn(source, key, bucket, context):
             name = keysplit[3]
             elbname = name.replace(".", "/")
             if len(idsplit) > 1:
-                idvalue = idsplit[1]
                 partition = get_partition_from_region(region)
-                return "arn:{}:elasticloadbalancing:{}:{}:loadbalancer/{}".format(
-                    partition, region, idvalue, elbname
-                )
+                pattern = re.compile("^[0-9]{12}$") #regex to check for 12-digits only - AWS account id is 12 digit
+                if bool(pattern.match(idsplit[1])): #checking that id is 12-digit
+                    idvalue = idsplit[1]
+                    return "arn:{}:elasticloadbalancing:{}:{}:loadbalancer/{}".format(partition, region, idvalue, elbname)
```

Comment:
You can leave the `return` statement after the if/elif clause, since it's the same in both branches of the code